process/smartContract/process.go

```diff
@@ -258,6 +263,11 @@ func (sc *scProcessor) ExecuteSmartContractTransaction(
 		return nil
 	}
 
+	err = sc.saveAccounts(acntSnd, acntDst)
```

Comment:
I would do saveAccount only where it is needed. acntSnd might not be in current shard.

Reply:
acntSnd is altered in the processScPayment() func, so it has to be saved here, because the next func (processVMOutput) might reload the account, so it has to be saved before reloading.
If acntSnd is nil, saveAccounts() func will not save the account.